python/lsst/afw/cameraGeom/cameraFactory.py

```diff
@@ -48,6 +48,28 @@ def makeDetector(detectorConfig, ampInfoCatalog, focalPlaneToPupil):
         transforms,
     )
 
+def copyDetector(detector, ampInfoCatalog=None):
```

Comment:
This isn't the only time that I've wanted to modify a Detector (e.g. to update gains while investigating TS-8 data), so I think that the ability to clone a Detector is valuable.